src/mem/cache/prefetch/xs_stream.cc

```
@@ -124,15 +135,16 @@ XsStreamPrefetcher::sendPFWithFilter(const PrefetchInfo &pfi, Addr addr, std::ve
         // Count generated prefetch
         prefetchStats.pfGenerated++;
 
-        if (filter->contains(pf_addr)) {
+        Addr filter_key = sharedFilterKey(pfi, pf_addr);
```

Cppcheck (2.21.0)

[error] Shifting 64-bit value by 64 bits is undefined behaviour

(shiftTooManyBits)